### PR TITLE
feat: delete `masked_cast`

### PR DESCRIPTION
**Pull request by JonathanStarup**
it is veeery brittle and both requires - and it broken by - inlining

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/language/phase/Desugar.scala [hunk 1]
```diff
@@ -1541,8 +1537,7 @@ object Desugar {
     val e = visitExp(exp0)
     val prefix = mkDebugPrefix(e, kind0, loc0)
     val e1 = DesugaredAst.Expr.Cst(Constant.Str(prefix), loc0)
-    val call = mkApplyFqn("Debug.debugWithPrefix", List(e1, e), loc0)
-    DesugaredAst.Expr.UncheckedMaskingCast(call, loc0)
+    mkApplyFqn("Debug.debugWithPrefix", List(e1, e), loc0)
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/Desugar.scala:1540]: Does this require a cast or no?

**Comment by magnus-madsen** [comment]
@JonathanStarup Merging, but do answer my Q

File: main/src/ca/uwaterloo/flix/language/phase/Desugar.scala (hunk 1 above)
Comment (JonathanStarup) [reply main/src/ca/uwaterloo/flix/language/phase/Desugar.scala:1540]: I've changed debugWithPrefix to be pure and have a cast inside .. but actually maybe that's a bad idea

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/Desugar.scala:1540]: Thats seems worse to me.

**Pull request merged**